### ISSUE TITLE
LSP support

### ISSUE THREAD

**Issue by ilovecherries** [y-ack/e#18]
a lot of useful features for a language including contextual autocomplete can be satisfied with an LSP and it's bad to deny that

https://github.com/gluon-lang/lsp-types
https://microsoft.github.io/language-server-protocol/

LSP usage will of course be optional but I think it would be a good idea if we had LSP support integrated into the editor itself
whether this should be implemented in Lua or in Rust will probably be considered if we ever get around to this

**Comment by y-ack** [comment]
yeah, this is probably important for going from 'you can type text' to 'actually a good programming environment'
lsp feels pretty heavy, but it might be the best option. will decide when there are lua modes that can consume completion info and stuff.
